addons/hr_holidays/models/mail_message.py

```diff
@@ -0,0 +1,10 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class MailMessage(models.Model):
+    _inherit = 'mail.message'
+
+    def _author_to_store_partner_fields(self):
+        return super()._author_to_store_partner_fields() + ["im_status", "leave_date_to"]
```

Comment:
Probably `leave_date_to` not necessary so could be removed to not increase query counts significantly.
I added `im_status` there because without it we wouldn't see the plane im status on conversation unless some other side-effect to fetch like manually open avatar card, which kills the purpose of seeing people on leave very quickly from message list.